Sources/MapboxDirections/Match/MBMatch.swift

```diff
@@ -1,114 +0,0 @@
-        self.confidence = confidence
-        self.tracepoints = tracepoints
-        self.waypointIndices = waypointIndices
-        super.init(legs: legs, distance: distance, expectedTravelTime: expectedTravelTime, coordinates: coordinates, speechLocale: speechLocale, options: matchOptions)
-    }
-    
-    /**
-     Initializes a new match object with the given JSON dictionary representation and tracepoints.
-     
-     - parameter json: A JSON dictionary representation of the route as returned by the Mapbox Map Matching API.
-     - parameter tracepoints: An array of `Tracepoint` that the match found in order.
-     - parameter matchOptions: The `MatchOptions` used to create the request.
-    */
-    @objc(initWithJSON:tracepoints:waypointIndices:matchOptions:)
-    public convenience init(json: [String: Any], tracepoints: [Tracepoint], waypointIndices: IndexSet, matchOptions: MatchOptions) {
-        let legInfo = zip(zip(tracepoints.prefix(upTo: tracepoints.endIndex - 1), tracepoints.suffix(from: 1)),
-                          json["legs"] as? [JSONDictionary] ?? [])
-        let legs = legInfo.map { (endpoints, json) -> RouteLeg in
-            return RouteLeg(json: json, source: endpoints.0, destination: endpoints.1, options: RouteOptions(matchOptions: matchOptions))
-        }
-        
-        let distance = json["distance"] as! Double
-        let expectedTravelTime = json["duration"] as! Double
-        
-        let coordinates = matchOptions.shapeFormat.coordinates(from: json["geometry"])
-        
-        let confidence = (json["confidence"] as! NSNumber).floatValue
-        
-        var speechLocale: Locale?
-        if let locale = json["voiceLocale"] as? String {
-            speechLocale = Locale(identifier: locale)
-        }
-        
-        self.init(matchOptions: matchOptions, legs: legs, tracepoints: tracepoints, distance: distance, expectedTravelTime: expectedTravelTime, coordinates: coordinates, confidence: confidence, speechLocale: speechLocale, waypointIndices: waypointIndices)
-    }
-    
-    /**
-     A number between 0 and 1 that indicates the Map Matching API’s confidence that the match is accurate. A higher confidence means the match is more likely to be accurate.
-     */
-    @objc open var confidence: Float
-    
-    
-    /**
-     Tracepoints on the road network that match the tracepoints in the match options.
-     
-     Any outlier tracepoint is omitted from the match. This array represents an outlier tracepoint is a `Tracepoint` object whose `Tracepoint.coordinate` property is `kCLLocationCoordinate2DInvalid`.
-     */
-    @objc open var tracepoints: [Tracepoint]
-    
-    
-    /**
-     Index of the waypoint inside the matched route.
-     */
-    @objc open var waypointIndices: IndexSet?
```

Comment:
This property is distinct from the `DirectionsOptions.waypointIndices` property that was justifiably removed. This property is meant to indicate which of the items in `tracepoints` is a (leg-separating) waypoint. If we remove this property, then we need to make sure the relevant tracepoints’ `separatesLegs` properties are set.